lib/synapse/haproxy.rb

```diff
@@ -1030,19 +1030,33 @@ def generate_backend_stanza(watcher, config)
       ]
     end
 
+    def haproxy_exec(command)
+      info = nil
+      s = UNIXSocket.new(@opts['socket_file_path'])
+      s.write(command)
+      info = s.read
+    rescue StandardError => e
+      log.warn "synapse: unhandled error reading stats socket: #{e.inspect}"
+      @restart_required = true
+      nil
+    ensure
+      unless s.nil?
+        begin
+          s.close
+        rescue StandardError => e
+          log.warn "synapse: unhandled error closing stats socket: #{e.inspect}"
+        end
+      end
+      info
```

Comment:
if you're thinking that this will return `info` from the function since the `ensure` block is the last one that runs, that's incorrect. this does nothing, and can be omitted. example:
```
irb(main):001:0> a = begin
irb(main):002:1* 5
irb(main):003:1> ensure
irb(main):004:1* 'hi'
irb(main):005:1> end
=> 5
irb(main):006:0> a
=> 5
```